From 3fe56f60b893c4fd82ab72f296e895d4e928e021 Mon Sep 17 00:00:00 2001
From: Mikhail Tsukerman <mikhail.tsukerman@advantech.de>
Date: Mon, 19 Jan 2026 17:00:22 +0100
Subject: [PATCH] Revert "serial: imx: Restore original RXTL for console to fix
 data loss"

20260313: update for lf-6.18.2-1.0.0

Upstream-Status: Inappropriate

This reverts commit 765af231964d6a6e0ad6946a247d0b4d43d0242f.
---
 drivers/tty/serial/imx.c | 18 +++++-------------
 1 file changed, 5 insertions(+), 13 deletions(-)

diff --git a/drivers/tty/serial/imx.c b/drivers/tty/serial/imx.c
index 611e338a6c1f..d89ffc965b6f 100644
--- a/drivers/tty/serial/imx.c
+++ b/drivers/tty/serial/imx.c
@@ -235,8 +235,6 @@ struct imx_port {
 	enum imx_tx_state	tx_state;
 	struct hrtimer		trigger_start_tx;
 	struct hrtimer		trigger_stop_tx;
-	unsigned int		rxtl;
-
 	struct pm_qos_request   pm_qos_req;
 };
 
@@ -1342,7 +1340,6 @@ static void imx_uart_clear_rx_errors(struct imx_port *sport)
 
 #define TXTL_DEFAULT 8
 #define RXTL_DEFAULT 8 /* 8 characters or aging timer */
-#define RXTL_CONSOLE_DEFAULT 1
 #define TXTL_DMA 8 /* DMA burst setting */
 #define RXTL_DMA 9 /* DMA burst setting */
 
@@ -1468,7 +1465,7 @@ static void imx_uart_disable_dma(struct imx_port *sport)
 	ucr1 &= ~(UCR1_RXDMAEN | UCR1_TXDMAEN | UCR1_ATDMAEN);
 	imx_uart_writel(sport, ucr1, UCR1);
 
-	imx_uart_setup_ufcr(sport, TXTL_DEFAULT, sport->rxtl);
+	imx_uart_setup_ufcr(sport, TXTL_DEFAULT, RXTL_DEFAULT);
 
 	sport->dma_is_enabled = 0;
 }
@@ -1493,12 +1490,7 @@ static int imx_uart_startup(struct uart_port *port)
 		return retval;
 	}
 
-	if (uart_console(&sport->port))
-		sport->rxtl = RXTL_CONSOLE_DEFAULT;
-	else
-		sport->rxtl = RXTL_DEFAULT;
-
-	imx_uart_setup_ufcr(sport, TXTL_DEFAULT, sport->rxtl);
+	imx_uart_setup_ufcr(sport, TXTL_DEFAULT, RXTL_DEFAULT);
 
 	/* disable the DREN bit (Data Ready interrupt enable) before
 	 * requesting IRQs
@@ -1965,7 +1957,7 @@ static int imx_uart_poll_init(struct uart_port *port)
 	if (retval)
 		clk_disable_unprepare(sport->clk_ipg);
 
-	imx_uart_setup_ufcr(sport, TXTL_DEFAULT, sport->rxtl);
+	imx_uart_setup_ufcr(sport, TXTL_DEFAULT, RXTL_DEFAULT);
 
 	uart_port_lock_irqsave(&sport->port, &flags);
 
@@ -2057,7 +2049,7 @@ static int imx_uart_rs485_config(struct uart_port *port, struct ktermios *termio
 		/* If the receiver trigger is 0, set it to a default value */
 		ufcr = imx_uart_readl(sport, UFCR);
 		if ((ufcr & UFCR_RXTL_MASK) == 0)
-			imx_uart_setup_ufcr(sport, TXTL_DEFAULT, sport->rxtl);
+			imx_uart_setup_ufcr(sport, TXTL_DEFAULT, RXTL_DEFAULT);
 		imx_uart_start_rx(port);
 	}
 
@@ -2242,7 +2234,7 @@ imx_uart_console_setup(struct console *co, char *options)
 	else
 		imx_uart_console_get_options(sport, &baud, &parity, &bits);
 
-	imx_uart_setup_ufcr(sport, TXTL_DEFAULT, sport->rxtl);
+	imx_uart_setup_ufcr(sport, TXTL_DEFAULT, RXTL_DEFAULT);
 
 	retval = uart_set_options(&sport->port, co, baud, parity, bits, flow);
 
